incidents/2021-06-07-elifesciences-org-cannot-be-resolved.md

```diff
@@ -0,0 +1,69 @@
+
+- 12:10 Peter Rodgers confirms that he is seeing the same issue with accessing the Journal.
+
+- 12:12 After checking the holiday calendar in Breathe HR, Joel realises that Giorgio is on holiday and asks for some help from Daniel H instead.
+
+- 12:13 Daniel H joins Joel in a Google Meet and together they start to investigate the problem.
+
+- 12:20 Joel and Daniel use CloudTrail to view the events, and can see that the above operation [did indeed make some adjustments to DNS records](https://console.aws.amazon.com/cloudtrail/home?region=us-east-1#/events/a8877f8f-c35f-463b-bd10-5feaafb7c9340).
+
+- 12:25 After some investigation, Daniel H runs `./bldr update_infrastructure:journal--prod` with the intent to bring resources in line with the desired state defined in builder. The command succeeds, but as far as builder was concerned the infrastructure is in the expected state and hence it was a no-op.
+
+- 12:30 Next hunch is that the issue might be in Fastly, but without access to the 2FA phone neither Daniel or Joel can login.
+
+- 12:36 Joel emails Luke to see if he is available to join the debugging effort.
+
+- 12:40 Luke joins the debugging effort.
+
+- 12:43 Confirm that EC2 instances and LoadBalancers are all OK.
+
+- 12:45 Luke can access Fastly, and after an initial look the configuration looks OK.
+
+- 12:50 We start to build a picture of how the DNS is setup and how traffic should be routed, Route53 -> Fastly -> EC2 Load Balancers -> journal--prod-(1/2/3).
+
+- 12:55 We confirm in Route53 that there is no entry for `elifesciences.org`.
+
+- 13:00 We re-review the CloudTrail events and CloudFormation templates and confirm that the DNS entry for `elifesciences.org` has been deleted from Route53.
+
+- 13:07 We manually recreate the A record in Route53 for `elifesciences.org` based on the information in the CloudFormation template, Setting it to point at the IP addresses for Fastly.
+
+- 13:09 Confirm that `elifesciences.org` is resolving correctly and the Journal is back up.
+
+- 13:14 Joel posts [a message](https://elifesciences.slack.com/archives/C025LBBR4/p1623068061016300) in the #general channel in slack to let everybody know about the incident, confirming that it is resolved and that in some cases it might take up to an hour for DNS to update.
+
+## Contributing Factor(s)
+
+- Lack of understanding/experience around launching ad hoc instances using `builder`. In this specific case, it probably would've been wise to review what resources would be impacted by trying to launch a new instance of journal.
```

Comment:
My understanding of the current `prod` configuration for `journal` and it's safety for use was incorrect and I gave bad advice because of it. I probably gave off false confidence about it's use as well.

> it probably would've been wise to review what resources would be impacted by trying to launch a new instance of journal.

If you mean eyeball the configuration before going ahead, then possibly. I'm not convinced you would have seen anything to give you pause though - you're just not as familiar with builder's project file.

If you mean review the generated Cloudformation/Terraform file before use then ... I'm not sure it's actually displayed. I know something is displayed on *update*, but not *create*.

Reply:
> If you mean eyeball the configuration before going ahead, then possibly. I'm not convinced you would have seen anything to give you pause though - you're just not as familiar with builder's project file.

I agree, it's extremely unlikey i'd have spotted anything that would have made me pause for thought.